src/content/app/species-selector/views/species-selector-results-view/SpeciesSelectorResultsView.tsx

```diff
@@ -73,19 +74,20 @@ const Content = (props: { onClose: () => void }) => {
     for (const addedGenome of genomes) {
       trackAddedGenome(addedGenome);
     }
+
+    navigate(-1);
   };
 
-  return modalView === 'species-search' ? (
+  return searchParams.has('query') ? (
     <GenomeSelectorBySearchQuery
-      query={query}
+      query={searchParams.get('query') || ''}
       onSpeciesAdd={onSpeciesAdd}
       onClose={props.onClose}
     />
-  ) : selectedPopularSpecies ? (
+  ) : searchParams.has('species_taxonomy_id') ? (
     <GenomeSelectorBySpeciesTaxonomyId
       onSpeciesAdd={onSpeciesAdd}
-      speciesTaxonomyId={selectedPopularSpecies.species_taxonomy_id}
-      speciesImageUrl={selectedPopularSpecies.image}
+      speciesTaxonomyId={searchParams.get('species_taxonomy_id') || ''}
```

Comment:
` || ''` -> `as string` (you have checked that `searchParams.get('species_taxonomy_id')` exists above)